aws/docker/worker/generate_profile_plot.py

```diff
@@ -33,64 +33,122 @@
     }
 }
 
-def generate_profile_plot(thredds_url):
+def generate_profile_plot(erddap_dataset):
     '''
     Plot the parameters for a deployment
-    :param str thredds_url: THREDDS DAP endpoint
+    :param str erddap_dataset: the deployment name example:'ce_311-20200708T1723'
     '''
-    deployment_name = thredds_url.split('/')[7]
-
-    with netCDF4.Dataset("{}#fillmismatch".format(thredds_url)) as nc:
-        for parameter in PARAMETERS:
-            filename = '{}/{}.png'.format(deployment_name, parameter)
-            try:
-                plot_from_nc(deployment_name, nc, parameter, filename)
-            except Exception:
-                logging.exception("Failed to generate plot for {}".format(parameter))
-                traceback.print_exc()
-                continue
+    path = dir_path(erddap_dataset)
+    df = get_erddap_data(erddap_dataset)
+
+    for parameter in PARAMETERS:
+        title = erddap_dataset + ' ' + parameter[0].upper() + parameter[1:] + ' Profiles'
+        filename = '{}/{}.png'.format(path, parameter)
+        try:
+            plot_from_pd(title, df, parameter, filename)
+        except Exception:
+            logging.exception("Failed to generate plot for {}".format(parameter))
+            traceback.print_exc()
+            continue
     return 0
 
 
-def get_times(nc, x, z):
+def get_erddap_data(dataset_id):
+    '''
+    :param dataset_id: the deployment name example:'ce_311-20200708T1723'
+    :return: pandas DataFrame with deployment variable values
     '''
-    Converts an array of timestamps to the matplotlib epoch and builds a
-    meshgrid of the timestamps for each profile
-    :param netCDF4.Dataset nc: An open netCDF file
-    :param numpy.ndarray x: An array of time values
-    :param numpy.ndarray z: A multi-dimensional array representing the profiles
+    e = ERDDAP(
+      server='https://gliders.ioos.us/erddap',
+      protocol='tabledap',
+    )
+    e.response = 'csv'
+    e.dataset_id = dataset_id
+    e.variables = [
+                    'depth',
+                    'latitude',
+                    'longitude',
+                    'salinity',
+                    'temperature',
+                    'conductivity',
+                    'density',
+                    'time',
```

Comment:
Moving forward we're going to want to plot more variables such as oxygen, pH, CDOM, etc. Those variables dont always have the same variable name. We will likely need a function that generates this list of variables based on what's available in the dataset (using standard names).